crates/services/sync/src/import.rs

```diff
@@ -250,6 +230,35 @@ where
         .in_current_span()
         .await
     }
+
+    async fn get_block_for_header(
```

Comment:
You have some intersection with @bvrooman here, so maybe it is better not to update `fuel-core-sync` for now and only add new functionality on the p2p layer to avoid conflicts.

Reply:
Hmm. Okay. Most of the work I've been doing today has been in understanding the sync stuff. There actually is very little p2p stuff to do, turns out.
I've also been rewriting some of the tests and over in sync. But sounds like I should cut that out.
I'll look into @bvrooman 's work before moving forward.

Reply:
It looks like there is overlap in the `get_header_range` area.